dspace-api/src/main/java/org/dspace/importer/external/arxiv/metadatamapping/contributor/ArXivIdMetadataContributor.java

```diff
@@ -0,0 +1,187 @@
+                                           MetadataFieldConfig field) {
+        this.query = query;
+        this.prefixToNamespaceMapping = prefixToNamespaceMapping;
+        this.field = field;
+    }
+
+    /**
+     * Empty constructor for ArXivIdMetadataContributor
+     */
+    public ArXivIdMetadataContributor() {
+
+    }
+
+    private String query;
+
+    /**
+     * Return the MetadataFieldConfig used while retrieving MetadatumDTO
+     *
+     * @return MetadataFieldConfig
+     */
+    public MetadataFieldConfig getField() {
+        return field;
+    }
+
+    /**
+     * Setting the MetadataFieldConfig
+     *
+     * @param field MetadataFieldConfig used while retrieving MetadatumDTO
+     */
+    @Required
+    public void setField(MetadataFieldConfig field) {
+        this.field = field;
+    }
+
+    /**
+     * Return query used to create an xpathExpression on, this query is used to
+     *
+     * @return the query this instance is based on
+     */
+    public String getQuery() {
+        return query;
+    }
+
+    @Required
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    /**
+     * Retrieve the metadata associated with the given object.
+     * Depending on the retrieved node (using the query), different types of values will be added to the MetadatumDTO
+     * list
+     *
+     * @param t A class to retrieve metadata from.
+     * @return a collection of import records. Only the identifier of the found records may be put in the record.
+     */
+    @Override
+    public Collection<MetadatumDTO> contributeMetadata(OMElement t) {
+        List<MetadatumDTO> values = new LinkedList<>();
+        try {
+            AXIOMXPath xpath = new AXIOMXPath(query);
+            for (String ns : prefixToNamespaceMapping.keySet()) {
+                xpath.addNamespace(prefixToNamespaceMapping.get(ns), ns);
+            }
+            List<Object> nodes = xpath.selectNodes(t);
+            for (Object el : nodes) {
+                if (el instanceof OMElement) {
+                    values.add(metadataFieldMapping.toDCValue(field, ((OMElement) el).getText()));
+                } else if (el instanceof OMAttribute) {
+                    values.add(metadataFieldMapping.toDCValue(field, ((OMAttribute) el).getAttributeValue()));
+                } else if (el instanceof String) {
+                    values.add(metadataFieldMapping.toDCValue(field, (String) el));
+                } else if (el instanceof OMText) {
+                    values.add(metadataFieldMapping.toDCValue(field, ((OMText) el).getText()));
+                } else {
+                    System.err.println("node of type: " + el.getClass());
+                }
+            }
+            parseValue(values);
+            return values;
+        } catch (JaxenException e) {
+            System.err.println(query);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void parseValue(List<MetadatumDTO> dtos) {
```

Comment:
Since this is a new method compared to the SimpleXpathMetadatumContributor, can add some docs explaining what it does?

Reply:
I agree this new method should have JavaDocs. It's not clear the purpose of this method...it seems to be changing the current value of all MetadatumDTO objects?

Reply:
Thanks for adding the JavaDocs, I do wonder how the item id is used here, I thought this was about the ArXiv ID